File: backend_kishan/app/Models/IncomeEntry.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Database\Eloquent\Model;
use Illuminate\Database\Eloquent\Relations\BelongsTo;

/**
 * @property int $id
 * @property int $user_id
 * @property int $land_id
 * @property string $income_type
 * @property string $amount
 * @property \Illuminate\Support\Carbon|null $entry_date
 * @property string|null $note
 * @property string|null $bill_photo_path
 * @property string|null $bill_photo_mime
 * @property \Illuminate\Support\Carbon|null $created_at
 * @property \Illuminate\Support\Carbon|null $updated_at
 */
class IncomeEntry extends Model
{
    use HasFactory;

    protected $fillable = [
        'user_id',
        'land_id',
        'income_type',
        'amount',
        'entry_date',
        'note',
        'bill_photo_path',
        'bill_photo_mime',
    ];

    protected function casts(): array
    {
        return [
            'amount' => 'decimal:2',
            'entry_date' => 'date',
        ];
    }

    public function user(): BelongsTo
    {
        return $this->belongsTo(User::class);
    }

    public function land(): BelongsTo
    {
        return $this->belongsTo(Land::class);
    }
}

File: lib/models/income_entry.dart
import 'dart:typed_data';

class IncomeEntry {
  String type;
  double amount;
  String date;
  String note;
  String? billPhotoPath;
  Uint8List? billPhotoBytes;

  IncomeEntry({
    required this.type,
    this.amount = 0,
    required this.date,
    required this.note,
    this.billPhotoPath,
    this.billPhotoBytes,
  });

  IncomeEntry copyWith({
    String? type,
    double? amount,
    String? date,
    String? note,
    String? billPhotoPath,
    Uint8List? billPhotoBytes,
    bool clearBillPhoto = false,
  }) {
    return IncomeEntry(
      type: type ?? this.type,
      amount: amount ?? this.amount,
      date: date ?? this.date,
      note: note ?? this.note,
      billPhotoPath: clearBillPhoto
          ? null
          : (billPhotoPath ?? this.billPhotoPath),
      billPhotoBytes: clearBillPhoto
          ? null
          : (billPhotoBytes ?? this.billPhotoBytes),
    );
  }
}
